sql-plugin/src/main/java/com/nvidia/spark/rapids/RapidsHostColumnVector.java

```diff
@@ -143,7 +150,27 @@ public ColumnarArray getArray(int rowId) {
 
   @Override
   public ColumnarMap getMap(int ordinal) {
-    throw new IllegalStateException("Maps are currently not supported by rapids cudf");
+    ai.rapids.cudf.ColumnViewAccess<HostMemoryBuffer> structHcv = cudfCv.getChildColumnViewAccess(0);
+    // keys
+    ai.rapids.cudf.ColumnViewAccess<HostMemoryBuffer> firstHcv = structHcv.getChildColumnViewAccess(0);
+    HostColumnVectorCore firstHcvCore = new HostColumnVectorCore(firstHcv.getDataType(),
+        firstHcv.getRowCount(), Optional.of(firstHcv.getNullCount()), firstHcv.getDataBuffer(),
+        firstHcv.getValidityBuffer(), firstHcv.getOffsetBuffer(),
+        new ArrayList<HostColumnVectorCore>());
+    // values
+    ai.rapids.cudf.ColumnViewAccess<HostMemoryBuffer> secondHcv = structHcv.getChildColumnViewAccess(1);
+    HostColumnVectorCore secondHcvCore = new HostColumnVectorCore(secondHcv.getDataType(),
+        secondHcv.getRowCount(), Optional.of(secondHcv.getNullCount()), secondHcv.getDataBuffer(),
+        secondHcv.getValidityBuffer(), secondHcv.getOffsetBuffer(),
+        new ArrayList<HostColumnVectorCore>());
+
+    RapidsHostColumnVectorCore firstChild = new RapidsHostColumnVectorCore(
```

Comment:
Does this change look ok now that we are not copying buffers?

Reply:
This is fine.  If we wanted to go a bit overboard we could look at caching `firstChild` and `secondChild` some how.  Because this is going to be called in a loop and they will be recreated each time.  But for now we can probably ignore this and not worry about it.

Reply:
https://github.com/NVIDIA/spark-rapids/issues/1004